morepath/tests/test_json_obj.py

```diff
@@ -25,6 +25,30 @@ def dump_model_json(self, request):
     assert response.json == {'x': 'foo'}
 
 
+def test_json_obj_dump_app_arg():
+    class App(morepath.App):
+        pass
+
+    @App.path(path='/models/{x}')
+    class Model(object):
+        def __init__(self, x):
+            self.x = x
+
+    @App.json(model=Model)
+    def default(self, request):
+        return self
+
+    @App.dump_json(model=Model)
+    def dump_model_json(app, self, request):
```

Comment:
This signature with ``self`` as its second argument, although perfectly legal, goes against Python conventions.

I think that calling that formal argument ``model`` instead of ``self`` would make it follow Python conventions more closely, besides immediately showing the link with the dispatch argument in the "@App.dump_json" directive:

```
@App.dump_json(model=Model)
def dump_model_json(app, model, request):
    ...
```

This remarks applies to the examples in the docs as well.